pkg/api/tests.go

```diff
@@ -132,3 +154,129 @@ func PrintTestsJSON(release string, w http.ResponseWriter, req *http.Request, cu
 		sort(req).
 		limit(req))
 }
+
+func PrintTestsJSONFromDB(release string, w http.ResponseWriter, req *http.Request, dbc *db.DB) {
+	tests := testsAPIResult{}
+	var filter *Filter
+
+	queryFilter := req.URL.Query().Get("filter")
+	if queryFilter != "" {
+		filter = &Filter{}
+		if err := json.Unmarshal([]byte(queryFilter), filter); err != nil {
+			RespondWithJSON(http.StatusBadRequest, w, map[string]interface{}{"code": http.StatusBadRequest, "message": "Could not marshal query:" + err.Error()})
+			return
+		}
+	}
+
+	// If requesting a two day report, we make the comparison between the last
+	// period (typically 7 days) and the last two days.
+	period := req.URL.Query().Get("period")
+	if period != "current" && period != "twoDay" {
+		RespondWithJSON(http.StatusBadRequest, w, map[string]interface{}{"code": http.StatusBadRequest, "message": "Unknown period"})
+		return
+	}
+
+	testsResult, err := BuildTestsResults(dbc, release, period, filter)
+	if err != nil {
+		RespondWithJSON(http.StatusInternalServerError, w, map[string]interface{}{"code": http.StatusInternalServerError, "message": "Error building job report:" + err.Error()})
+		return
+	}
+
+	RespondWithJSON(http.StatusOK, w, testsResult.
+		sort(req).
+		limit(req))
+	RespondWithJSON(http.StatusOK, w, tests.
+		sort(req).
+		limit(req))
```

Comment:
We should only be rendering once here? The logs show an error:

```
2022/03/16 10:02:50 http: superfluous response.WriteHeader call from github.com/openshift/sippy/pkg/api.RespondWithJSON (api.go:335)
```

Reply:
Whoops!